File: frontend/src/components/WoD/Changeling/Skills.js
import React from 'react'
import FiveDots from '../FiveDots'
import './Skills.css'

export default class Skills extends React.Component
{
	render()
	{
		return (
			<div className="skills">
				<div className="mental">
					<div className="column">
						<div className="label">Academics:</div>
						<div className="label">Computer:</div>
						<div className="label">Crafts:</div>
						<div className="label">Investigation:</div>
						<div className="label">Medicine:</div>
						<div className="label">Occult:</div>
						<div className="label">Politics:</div>
						<div className="label">Science:</div>
					</div>
					<div className="column">
						<FiveDots className="dots academics" value={this.props.skills.mental.academics} />
						<FiveDots className="dots computer" value={this.props.skills.mental.computer} />
						<FiveDots className="dots crafts" value={this.props.skills.mental.crafts} />
						<FiveDots className="dots investigation" value={this.props.skills.mental.investigation} />
						<FiveDots className="dots medicine" value={this.props.skills.mental.medicine} />
						<FiveDots className="dots occult" value={this.props.skills.mental.occult} />
						<FiveDots className="dots politics" value={this.props.skills.mental.politics} />
						<FiveDots className="dots science" value={this.props.skills.mental.science} />
					</div>
				</div>
				<div className="physical">
					<div className="column">
						<div className="label">Athletics:</div>
						<div className="label">Brawl:</div>
						<div className="label">Drive:</div>
						<div className="label">Firearms:</div>
						<div className="label">Larceny:</div>
						<div className="label">Stealth:</div>
						<div className="label">Survival:</div>
						<div className="label">Weaponry:</div>
					</div>
					<div className="column">
						<FiveDots className="dots athletics" value={this.props.skills.physical.athletics} />
						<FiveDots className="dots brawl" value={this.props.skills.physical.brawl} />
						<FiveDots className="dots drive" value={this.props.skills.physical.drive} />
						<FiveDots className="dots firearms" value={this.props.skills.physical.firearms} />
						<FiveDots className="dots larceny" value={this.props.skills.physical.larceny} />
						<FiveDots className="dots stealth" value={this.props.skills.physical.stealth} />
						<FiveDots className="dots survival" value={this.props.skills.physical.survival} />
						<FiveDots className="dots weaponry" value={this.props.skills.physical.weaponry} />
					</div>
				</div>
				<div className="social">
					<div className="column">
						<div className="label">Animal Ken:</div>
						<div className="label">Empathy:</div>
						<div className="label">Expression:</div>
						<div className="label">Intimidation:</div>
						<div className="label">Persuasion:</div>
						<div className="label">Socialize:</div>
						<div className="label">Streetwise:</div>
						<div className="label">Subterfuge:</div>
					</div>
					<div className="column">
						<FiveDots className="dots animalKen" value={this.props.skills.social.animalKen} />
						<FiveDots className="dots empathy" value={this.props.skills.social.empathy} />
						<FiveDots className="dots expression" value={this.props.skills.social.expression} />
						<FiveDots className="dots intimidation" value={this.props.skills.social.intimidation} />
						<FiveDots className="dots persuasion" value={this.props.skills.social.persuasion} />
						<FiveDots className="dots socialize" value={this.props.skills.social.socialize} />
						<FiveDots className="dots streetwise" value={this.props.skills.social.streetwise} />
						<FiveDots className="dots subterfuge" value={this.props.skills.social.subterfuge} />
					</div>
				</div>
			</div>
		)
	}
}

File: src/gamesystems/cofd/data/skills/skills.rs
use serde::{Deserialize, Serialize};
use super::mental::SkillsMental;
use super::physical::SkillsPhysical;
use super::social::SkillsSocial;

#[derive(Clone, Copy, Debug, Default, Deserialize, PartialEq, Serialize)]
pub struct Skills
{
	pub academics: u32,
	pub animalKen: u32,
	pub athletics: u32,
	pub brawl: u32,
	pub computer: u32,
	pub crafts: u32,
	pub drive: u32,
	pub empathy: u32,
	pub expression: u32,
	pub firearms: u32,
	pub intimidation: u32,
	pub investigation: u32,
	pub larceny: u32,
	pub medicine: u32,
	pub occult: u32,
	pub persuasion: u32,
	pub politics: u32,
	pub science: u32,
	pub socialize: u32,
	pub stealth: u32,
	pub streetwise: u32,
	pub subterfuge: u32,
	pub survival: u32,
	pub weaponry: u32,
}

impl From<SkillsMental> for Skills
{
	fn from(value: SkillsMental) -> Self
	{
		return Self
		{
			academics: value.academics,
			computer: value.computer,
			crafts: value.crafts,
			investigation: value.investigation,
			medicine: value.medicine,
			occult: value.occult,
			politics: value.politics,
			science: value.science,
			..Default::default()
		};
	}
}

impl From<SkillsPhysical> for Skills
{
	fn from(value: SkillsPhysical) -> Self
	{
		return Self
		{
			athletics: value.athletics,
			brawl: value.brawl,
			drive: value.drive,
			firearms: value.firearms,
			larceny: value.larceny,
			stealth: value.stealth,
			survival: value.survival,
			weaponry: value.weaponry,
			..Default::default()
		};
	}
}

impl From<SkillsSocial> for Skills
{
	fn from(value: SkillsSocial) -> Self
	{
		return Self
		{
			animalKen: value.animalKen,
			empathy: value.empathy,
			expression: value.expression,
			intimidation: value.intimidation,
			persuasion: value.persuasion,
			socialize: value.socialize,
			streetwise: value.streetwise,
			subterfuge: value.subterfuge,
			..Default::default()
		};
	}
}

impl Skills
{
	pub fn updateMental(&mut self, value: SkillsMental)
	{
		self.academics = value.academics;
		self.computer = value.computer;
		self.crafts = value.crafts;
		self.investigation = value.investigation;
		self.medicine = value.medicine;
		self.occult = value.occult;
		self.politics = value.politics;
		self.science = value.science;
	}
	
	pub fn updatePhysical(&mut self, value: SkillsPhysical)
	{
		self.athletics = value.athletics;
		self.brawl = value.brawl;
		self.drive = value.drive;
		self.firearms = value.firearms;
		self.larceny = value.larceny;
		self.stealth = value.stealth;
		self.survival = value.survival;
		self.weaponry = value.weaponry;
	}
	
	pub fn updateSocial(&mut self, value: SkillsSocial)
	{
		self.animalKen = value.animalKen;
		self.empathy = value.empathy;
		self.expression = value.expression;
		self.intimidation = value.intimidation;
		self.persuasion = value.persuasion;
		self.socialize = value.socialize;
		self.streetwise = value.streetwise;
		self.subterfuge = value.subterfuge;
	}
}
